### PR TITLE
docs: create-react-app recipe

### PR DESCRIPTION
**Pull request by davidroeca**
This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] tests
- [x] documentation
- [ ] metadata

### Breaking Changes?

- [ ] yes
- [x] no

If yes, please describe the breakage.

### Please Describe Your Changes

Implementing a recipe to help guide usage of this plugin with a non-ejected `create-react-app` app.

Relates to https://github.com/shellscape/webpack-plugin-serve/issues/117

### CODE REVIEW

**Comment by shellscape** [comment]
Thanks for the PR! 🍺 

It looks like the tests are failing because `ava` is seeing `*.test.s` files in the recipes directory. To fix that, please update the `ava` config in `package.json` so that it looks like:

```json
  "ava": {
    "files": [
      "!**/fixtures/**",
      "!**/helpers/**",
      "!**/recipes/**"
    ]
  },
```

That should allow CI to pass

File: recipes/create-react-app/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+## 🍲 Create-React-App App
+
+Since it was built to hide configuration from developers, `create-react-app`
+actually makes it harder to use great plugins like this one.
+
+### ☠️ Warning ☠️
+
+This recipe is definitely neither vegan nor gluten-free. It can neither be
```
Comment (shellscape) [inline recipes/create-react-app/README.md]: 😂

File: recipes/create-react-app/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+## 🍲 Create-React-App App
+
+Since it was built to hide configuration from developers, `create-react-app`
+actually makes it harder to use great plugins like this one.
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion
actually makes it harder to use great plugins like this one. In this recipe, we'll show how to use `create-react-app` with `webpack-plugin-serve` without having to eject.
```

File: recipes/create-react-app/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+This recipe may have broken features that the author doesn't use.
+
+If you're working on a new project, *it is strongly advised* that you consider
+one of the following alternatives:
+* Neutrinojs [9.0 beta release candidate](https://github.com/neutrinojs/neutrino/issues/1129)
+* Finding/creating a project template and using [degit](https://github.com/Rich-Harris/degit)
+* A custom webpack config
+
+🚨You have been warned. 🚨
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion
🚨 You have been warned. 🚨
```

File: recipes/create-react-app/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Since it was built to hide configuration from developers, `create-react-app`
+actually makes it harder to use great plugins like this one.
+
+### ☠️ Warning ☠️
+
+This recipe is definitely neither vegan nor gluten-free. It can neither be
+confirmed nor denied whether it contains traces of lead or arsenic. Develop at
+your own risk.
+
+This recipe may have broken features that the author doesn't use.
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion
If you find that implementing this recipe causes unintended side effects or problems, please open a new [Documentation Issue](https://github.com/shellscape/webpack-plugin-serve/issues/new?template=DOCS.md).
```

File: recipes/create-react-app/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+With that out of the way, let's use `webpack-plugin-serve` with
+`create-react-app`.
+
+Note that this entire section has already been taken care of, and the artifacts
+now live within this codebase. However, in order for you to effectively
+replicate what has been created here, it will help to write down these steps.
+
+You'll need to create a new app using the latest version of `create-react-app`:
+
+```bash
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion
```console
```

better syntax highlighting

File: recipes/create-react-app/README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+# Install react-hot-loader dependencies
+npm install -D @hot-loader/react-dom
+npm install react-hot-loader
+```
+
+Under the hood, `create-react-app` uses the package `react-scripts` to do most
+of the heavy lifting. We'll be overriding its
+[configuration files](https://github.com/facebook/create-react-app/tree/master/packages/react-scripts/config)
+using [react-app-rewired](https://github.com/timarney/react-app-rewired).
+Note that these configurations change between versions of
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion

_Note: that these configurations change between versions of
```

File: recipes/create-react-app/README.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+* A custom webpack config
+
+🚨You have been warned. 🚨
+
+## Recipe
+
+With that out of the way, let's use `webpack-plugin-serve` with
+`create-react-app`.
+
+Note that this entire section has already been taken care of, and the artifacts
```
Comment (shellscape) [inline recipes/create-react-app/README.md]:
```suggestion
_Note: that this entire section has already been taken care of, and the artifacts
```

**Comment by davidroeca** [comment]
@shellscape thanks for the review! Should be cleaned up, let me know if anything else should change

**Pull request merged**